tsconfig.json

```diff
@@ -0,0 +1,23 @@
+{
+	"compilerOptions": {
+		"allowJs": true,
+		"esModuleInterop": true,
+		"isolatedModules": true,
+		"jsx": "react-jsx",
+		"lib": [ "dom", "esnext" ],
+		"module": "esnext",
+		"moduleResolution": "node",
+		"noEmit": true,
+		"noFallthroughCasesInSwitch": true,
+		"resolveJsonModule": true,
+		"skipLibCheck": true,
+		"strict": false,
+		"target": "ES6",
+		"paths": {
+			// e.g. "@automattic/jetpack-components"
+			// will resolve to "projects/js-packages/components"
+			"@automattic/jetpack-*": [ "./projects/js-packages/*" ]
```

Comment:
This should no longer be needed after #23823. That you still have it makes me wonder how much else in this PR is unnecessary. Does this file even need to be here, or does it only exist for extending and could be elsewhere (e.g. `tools/js-tools/tsconfig.base.json`)?

Reply:
> That you still have it makes me wonder how much else in this PR is unnecessary.

I don't think there is anything unnecessary in this PR now.

> Does this file even need to be here, or does it only exist for extending and could be elsewhere (e.g. `tools/js-tools/tsconfig.base.json`)?

Not really, it's just for extending. Moving it to `js-tools` is an excellent suggestion. Thank you.

I have made the changes.